### PR TITLE
update spp; highlight boost and icu incompatibility

### PR DESCRIPTION
**Pull request by daler**
* [ ] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/r-spp/meta.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ requirements:
 test:
   commands:
     - $R -e "library('spp')"
-
+    - $R -e "library('spp'); read.bam.tags(system.file('extdata', 'ex1.bam', package='Rsamtools'))"
```
Comment (daler) [inline recipes/r-spp/meta.yaml]: @bgruening can you please advise on how I should quote this for running with involucro? I can't seem to get it right.

Comment (daler) [inline recipes/r-spp/meta.yaml]: ping @bgruening --  any advice for this?

Comment (bgruening) [inline recipes/r-spp/meta.yaml]: can you try "library(\"spp\"); read.bam.tags(system.file(\"extdata\", \"ex1.bam\", package=\"Rsamtools\"))" ... need to play with it

**Comment by bgruening** [comment]
@daler this seems to work, but OSX is failing.

**Comment by daler** [comment]
@bioconda/osx I'm trying to set `-ferror-limit=0` on OSX, but can't manage to get clang to see it when called via `R CMD --build .`.

The error is from too many warnings: https://travis-ci.org/bioconda/bioconda-recipes/jobs/266182626#L458. 

https://clang.llvm.org/docs/UsersManual.html#options-to-control-error-and-warning-messages says to use `-ferror-limit=0` to disable this. So far I've tried:

- `export CFLAGS=-ferror-limit=0`
- `export CXXFLAGS=-ferror-limit=0`
- `export CPPFLAGS=-ferror-limit=0`
- `R CMD --build . --configure-args="-ferror-limit=0"`

Any other ideas? I've skipped it on osx for now, but would like to avoid that if possible.

**Comment by yesimon** [comment]
Maybe not the cleanest approach but you might able to modify `src/Makevars.in` and then maybe run `autoreconf -i` before `R CMD --build ...`

**Comment by daler** [comment]
Thanks @yesimon, that got me farther! Any suggestions on how to handle these: https://travis-ci.org/bioconda/bioconda-recipes/jobs/266293534#L835. It's compiling fine on linux; is it failing here because clang is being too picky?

**Comment by yesimon** [comment]
Unless you want to fix upstream in spp, looks like you're going to have to use `gcc` https://github.com/hms-dbmi/spp/blob/master/src/maqread.cpp#L23

**Comment by yesimon** [comment]
Actually you might be able to get away with `-stdlib=libstdc++`

**Comment by daler** [comment]
@yesimon perfect -- `-stdlib=libstdc++` did the trick. Thanks for your help.

**Comment by yesimon** [comment]
@daler Just be aware that `libstdc++` is deprecated for clang and will not work for code that requires c++11 and later.

**Comment by daler** [comment]
Thanks. I'll open an issue upstream on this.

**Pull request merged**